### PR TITLE
Update Register Endpoint & Add Cookie Response Logic

### PR DESCRIPTION
**Pull request by freddyzhounyc**
Update register endpoint to employ validation checks for inputs. Add cookie embedding feature in response headers.

### CODE REVIEW

File: src/test/java/com/stayhome/healthcare/controllers/AuthControllerIntegrationTests.java [hunk 1]
```diff
@@ -90,10 +92,222 @@ public void testRegisterRegistersNonRegisteredUser() throws Exception {
...
+        ).andExpect(
+                MockMvcResultMatchers.status().isOk()
+        ).andExpect(
+                MockMvcResultMatchers.jsonPath("$.email").value(request.getEmail().trim())
+        ).andExpect(
+                MockMvcResultMatchers.jsonPath("$.username").value(request.getUsername().trim())
+        );
+    }
+    @Test
+    public void testThatRegisterDoesNotRegisterWithABadEmailA() throws Exception {
```
Comment (alpreston0419) [inline src/test/java/com/stayhome/healthcare/controllers/AuthControllerIntegrationTests.java]: Instead of naming test as _bad email a_, can you give a brief description of what kind of bad email is being tested (ie. testThatRegisterDoesNotRegisterWithABadEmailA -> testRegisterFailsWithEmailMissingDomain)

File: src/test/java/com/stayhome/healthcare/TestDataUtil.java [hunk 2]
```diff
@@ -12,23 +13,150 @@ public static RegisterRequest createRegisterRequestA1() {
...
+    }
+    public static RegisterRequest createRegisterRequestWithBadEmailB() {
+        return RegisterRequest.builder()
+                .username("myUser")
+                .email("myUsergmail.com")
+                .password("myPa$$w0rd")
+                .role(Role.PATIENT)
+                .build();
+    }
+    public static RegisterRequest createRegisterRequestWithBadEmailC() {
```
Comment (alpreston0419) [inline src/test/java/com/stayhome/healthcare/TestDataUtil.java]: How about a bad email missing the email domain entirely: myUser
Do we need a test for empty string?
